Mathlib/Analysis/Convex/Radon.lean

```diff
@@ -0,0 +1,232 @@
+/-
+Copyright (c) 2023 Vasily Nesterov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Vasily Nesterov
+-/
+import Mathlib.Analysis.Convex.Combination
+import Mathlib.LinearAlgebra.AffineSpace.Independent
+import Mathlib.Tactic.Linarith
+
+/-!
+# Radon's convexity theorem
+
+Radon's theorem on convex sets states that any affine dependent set can be partitioned into two sets
+whose convex hulls intersect.
+
+## Main results
+
+* `Radon_partition`: Radon convexity theorem
+
+## Tags
+convex hull, radon
+
+-/
+
+
+open Set Finset
+open BigOperators
+
+universe u
+
+variable {𝕜 : Type*} {E : Type u} [LinearOrderedField 𝕜] [AddCommGroup E] [Module 𝕜 E]
+
+/-- Any family `f` of affine dependent vectors contains a set `I` with the property that
+convex hulls of `I` and `Iᶜ` intersect. -/
+theorem radon_partition {ι : Type*} {f : ι → E}
+    (h : ¬AffineIndependent 𝕜 f) : ∃ (I : Set ι),
+    (Set.Nonempty ((convexHull 𝕜 (f '' I)) ∩ (convexHull 𝕜 (f '' Iᶜ)))) := by
+  unfold AffineIndependent at h; push_neg at h
+  rcases h with ⟨s1, w, h_wsum, h_vsum, nonzero_w_index, h1, h2⟩
+  let I : Finset ι := s1.filter (fun i => w i ≥ 0)
+  let J : Finset ι := s1.filter (fun i => w i < 0)
+  use (I : Set ι)
+
+  let weights_sum_I := ∑ i in I, w i
+  let weights_sum_J := ∑ i in J, w i
+  have h3 : weights_sum_I + weights_sum_J = 0 := by
+    rw [←h_wsum]; simp
+    have h3_aux := Finset.sum_filter_add_sum_filter_not s1 (fun i => 0 ≤ w i) w
+    simp at h3_aux; exact h3_aux
```

Comment:
```suggestion
    simpa only [h_wsum, not_le] using sum_filter_add_sum_filter_not s1 (fun i => 0 ≤ w i) w
```
Nothing fundamental here, but if you don't know about `simpa` you should learn about it, it's a great way to do "`exact` up to some rewriting/simplification"!